sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -638,8 +640,30 @@ def __init__(
     if max_batch_size is not None:
       self._batching_kwargs['max_batch_size'] = max_batch_size
     self._large_model = large_model
+
+    # Check if the device is specified twice. If true then the device parameter
+    # of model handler is overridden.
+    self._deduplicate_device_value(device)
     _validate_constructor_args_hf_pipeline(self._task, self._model)
 
+  def _deduplicate_device_value(self, device: str):
+    if 'device' not in self._load_pipeline_args:
+      if device == 'CPU':
+        self._load_pipeline_args['device'] = 'cpu'
+      else:
+        if is_gpu_available_torch():
+          self._load_pipeline_args['device'] = 'cuda:1'
+        else:
+          _LOGGER.warning(
+              "HuggingFaceModelHandler specified a 'GPU' device, "
+              "but GPUs are not available. Switching to CPU.")
+          self._load_pipeline_args['device'] = 'cpu'
+    else:
+      if device:
```

Comment:
Since `device` has a default value, this will always trigger. I think we should change `device`'s default value  to `device: Optional[str] = None`

Reply:
Oops I missed changing that to None while I did other changes keeping that in mind :p.